src/App.jsx

```diff
@@ -1,14 +1,71 @@
 import './App.css';
+import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+import ColorChoice from './components/ColorChoice';
+import { useState } from 'react';
+
+// count like feature
+const calculateTotalLikes = (messages) =>
+  messages.reduce((total, message) => total + (message.liked ? 1 : 0), 0);
 
 const App = () => {
+  const [messages, setMessages] = useState(chatMessages);
+  const [localColor, setLocalColor] = useState('color');
+  const [remoteColor, setRemoteColor] = useState('color');
+
+  // Determine local and remote participants dynamically
+  const localSender = messages.length > 0 ? messages[0].sender : 'Local';
+  const remoteSender = [...new Set(messages.map((msg) => msg.sender))]
+    .find((sender) => sender !== localSender) || 'Remote';
+
+  const totalLikes = calculateTotalLikes(messages);
+
+  const toggleLike = (id) => {
+    const updatedMessages = messages.map((msg) =>
+      msg.id === id ? { ...msg, liked: !msg.liked } : msg
+    );
+    setMessages(updatedMessages);
+  };
```

Comment:
Great job using the ternary for this relatively simplistic check! You make your function more succinct but still maintain readability.